### PR TITLE
GCD and inverses for great good.

### PR DESCRIPTION
**Pull request by h4tguy**

### CODE REVIEW

File: gcd-inverse/GCD.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			long ny = x%y;
+			long div = x/y;
+			long [][] ncosts ={
+				costs[1],
+				{mod(costs[0][0] - div * costs[1][0],orig_y), mod(costs[0][1] - div * costs[1][1],orig_y)}
+			};
+			x = nx;
+			y = ny;
+			costs = ncosts;
+			return reduce();
```
Comment (MrHamdulay) [inline gcd-inverse/GCD.java]: wouldn't this be easier as a while loop? It's not more complicated and reduces time spent doing method calls.

Comment (h4tguy) [inline gcd-inverse/GCD.java]: Sure. I guess I'm used to functional paradigms...

**Comment by h4tguy** [comment]
@MrHamdulay

**Comment by MrHamdulay** [comment]
lgtm

**Pull request merged**